GANDLF/cli/huggingface_hub_handler.py

```diff
@@ -0,0 +1,259 @@
+from huggingface_hub import HfApi, snapshot_download, ModelCardData, ModelCard
+from typing import List, Union
+from GANDLF import version
+from pathlib import Path
+from GANDLF.utils import get_git_hash
+
+readme_template = """
```

Comment:
Is this a simple copy of the model card template found [here](https://github.com/huggingface/huggingface_hub/blob/main/src/huggingface_hub/templates/modelcard_template.md)? If yes, I can suggest to either:
- directly reuse the template from `huggingface_hub` (i.e. `ModelCard.from_template(card_data)` without the `template_str`).
- or define your own template but in this case you should only put the relevant fields and descriptions for your library (instead of having all fields as empty)

Reply:
Thanks  @Wauplin for making me aware of this ,I will definitely go through it and make  required changes as you mentioned

Reply:
Hi @Wauplin,

We had an internal discussion on what would be the best way for us to showcase potential model uploaders with a specific set of required options for the model card. Thus far, we have landed on using a custom [model card](https://github.com/user-attachments/files/16994290/hf_model_card_template.md). The reason to have all the fields present is provide the ability for a user to put in more information than what we _require_. 

Here, we have put the string `"REQUIRED_FOR_GANDLF"` for the fields that are explicitly needed for the user to populate, and the rest have been left as present in the [template](https://github.com/huggingface/huggingface_hub/blob/main/src/huggingface_hub/templates/modelcard_template.md).

In the code, we plan to add 2 checks:

1. If `"REQUIRED_FOR_GANDLF"` is found, we present an error to the user saying that this field needs to be populated with appropriate information.
2. The `Repository` key should always be `https://github.com/mlcommons/GaNDLF`.

Thoughts?

Reply:
This seems a sensible idea to me yes!

Reply:
Brilliant, thanks for the confirmation! We'll get on it right away. 😄

Reply:
@sarthakpati  @Wauplin  so how can we  test this file if we propose the upload functionality as we only have entry points tests, do we have to mention a specific directory there

Reply:
Perhaps you can leverage one of the existing training tests to test the upload. I would recommend [this one](https://github.com/mlcommons/GaNDLF/blob/master/testing/test_full.py#L2171), since this would only upload a single model. 

Ensure you put an appropriate description for it (such as `Unit testing model` or something) to make it clear for anyone viewing it. Is there a way to update an existing model, @Wauplin?